lib/cli-engine/cli-engine.js

```diff
@@ -863,7 +863,10 @@ class CLIEngine {
         // Clear the last used config arrays.
         lastConfigArrays.length = 0;
 
-        if (resolvedFilename && this.isPathIgnored(resolvedFilename)) {
+        const dotfilesPattern = /(?:(?:^\.)|(?:[/\\]\.))[^/\\.].*/u;
```

Comment:
Is there a reason this logic can’t go into `isPathIgnored()`?

Reply:
We need the resolved filename to test the dot files pattern. That's why it has to go here.

Reply:
I don't think we should add `isDot` to the public API of `isPathIgnored()`, since that's something the method can calculate (as demonstrated here!). `isPathIgnored()` receives the resolved filename as argument, so we should be able to avoid changing the public API by moving this check in that method.

Reply:
https://github.com/eslint/eslint/blob/master/tests/lib/cli-engine/cli-engine.js#L3849
https://github.com/eslint/eslint/blob/master/tests/lib/cli-engine/cli-engine.js#L3841

These test cases are failing. However they sound counter intuitive. When ignore is false, we want the dot files to be not ignored right?

Reply:
My guess is that the intention is to avoid linting directories such as `.git/` when the `--no-ignore` option is set. I think we should scope this PR to the specific bug fix report not change that behavior in this PR (though we always welcome RFCs for core changes if you feel differently).

Reply:
They don't fail if the check is made within the execute on text function and isDot passed as an extra parameter to isPathIgnored

Reply:
@kaicataldo I have modified to code. Let me know if that will work.